file.go

```diff
@@ -0,0 +1,100 @@
+package sivafs
+
+import (
+	"io"
+	"sync"
+
+	"gopkg.in/src-d/go-billy.v2"
+	"gopkg.in/src-d/go-siva.v1"
+)
+
+type file struct {
+	billy.BaseFile
+
+	w siva.Writer
+	r *io.SectionReader
+	l *sync.Mutex
+
+	closeNotify func() error
+}
+
+func newFile(filename string, w siva.Writer,
+	l *sync.Mutex, closeNotify func() error) billy.File {
+
+	return &file{
+		BaseFile:    billy.BaseFile{BaseFilename: filename},
+		w:           w,
+		l:           l,
+		closeNotify: closeNotify,
+	}
+}
+
+func openFile(filename string, r *io.SectionReader, l *sync.Mutex) billy.File {
+	return &file{
+		BaseFile: billy.BaseFile{BaseFilename: filename},
+		r:        r,
+		l:        l,
+	}
+}
+
+func (f *file) Read(p []byte) (int, error) {
+	if f.r == nil {
+		return 0, ErrWriteOnlyFile
+	}
+
+	f.l.Lock()
```

Comment:
The same reason as you lock on the rest of the code, because if the lock is protecting things like `.Closed` or `.r`, then not locking on them will cause a race condition with a thread calling Close for instance.  In any case, I don't know what the lock is protecting, so this is your call and maybe something you should explain in some comments.